### PR TITLE
[GC stress] Add multiple DDSs to data stores in stress tests

### PR DESCRIPTION
**Pull request by agarwal-navin**
## Description
Added multiple DDSs to the leaf data stores. Updated the activity on it to changing none, one or both of them. This will give us coverage when between summaries some DDS re-summarize where as some use summary handles. With this change, we should have coverage of all these scenarios:
1. Data store changes and re-summarizes via a Summary tree.
2. Data store does not change and does incremental summary via Summary handle.
3. DDS in data store changes and re-summarizes via a Summary tree.
4. DDS in data store does not change and does incremental summary via Summary handle.

## Reviewer guidance
This is for an experimental stress test in test/gc-stress branch. This is not in the main FF branch and does not affect its stress test

### CODE REVIEW

File: packages/test/test-service-load/src/gcDataStores.ts [hunk 1]
```diff
@@ -195,6 +197,28 @@ export class DataObjectLeaf extends BaseDataObject implements IGCActivityObject
...
+		await super.initializingFirstTime();
+		this.root.set<IFluidHandle>(this.counter2Key, SharedCounter.create(this.runtime).handle);
+	}
+
+	protected async hasInitialized(): Promise<void> {
+		await super.hasInitialized();
+		const handle = this.root.get<IFluidHandle<SharedCounter>>(this.counter2Key);
+		assert(handle !== undefined, "The counter 2 handle should exist on initialization");
+		this._counter2 = await handle.get();
+	}
```
Comment (markfields) [inline packages/test/test-service-load/src/gcDataStores.ts:220]: Can we share this code with the code in base for the first counter? Not that it's much, but it's something. Could be worse to share it though in terms of over-engineering

File: packages/test/test-service-load/src/gcDataStores.ts [hunk 2]
```diff
@@ -92,15 +92,17 @@ const BlobActivityType = {
 type BlobActivityType = typeof BlobActivityType[keyof typeof BlobActivityType];
 
 /**
- * Activities that can be performed by the leaf blob object.
+ * Activities that can be performed by the leaf data object.
  */
 const LeafActivityType = {
 	/** Don't do anything. */
 	None: 0,
```
Comment (markfields) [inline packages/test/test-service-load/src/gcDataStores.ts:99]: Add anotherNone to give a little more weight to it to get more incremental coverage?

**Comment by markfields** [comment]
> Data store changes and re-summarizes via a Summary tree

Does "Data store changes" happen if either DDS changes?  Or does the Data store only change when DDSes are created or something?

**Review by markfields — APPROVED** [review]

File: packages/test/test-service-load/src/gcDataStores.ts (hunk 2 above)
Comment (agarwal-navin) [reply packages/test/test-service-load/src/gcDataStores.ts:99]: It's 33% already which is pretty good right. The last one `Count` is not an activity.

File: packages/test/test-service-load/src/gcDataStores.ts (hunk 1 above)
Comment (agarwal-navin) [reply packages/test/test-service-load/src/gcDataStores.ts:220]: Yeah, I feel like it will be over doing it. It's fine for now :-).

**Comment by agarwal-navin** [comment]
> > Data store changes and re-summarizes via a Summary tree
> 
> Does "Data store changes" happen if either DDS changes? Or does the Data store only change when DDSes are created or something?

It happens when it gets an op which includes any ops for its DDSes. So both of the scenarios you described above will result in data store changing.

**Pull request merged**